### PR TITLE
Enter key newline and validation

### PR DESCRIPTION
**Pull request by TitouanT**
This a fix to #12
MultiLine: Enter -> newline, Ctrl+Enter -> done
MonoLine: [Ctrl+]Enter -> done

This behavior is working for flutter >=0.10.1

note: by default it just loses focus of the input but you can personalize the action with onSubmitted callback.
see  https://github.com/flutter/flutter/blob/v0.10.1/packages/flutter/lib/src/widgets/editable_text.dart#L525

### CODE REVIEW

**Comment by Drakirus** [comment]
LGTM, I'll merge it in a few days.

**Comment by Drakirus** [comment]
For testing purposes, can you update the simpleDemo flutter app (dart side) to provide a `onSubmitted` callback to the `TextField`.

**Comment by TitouanT** [comment]
- [x] add an `onSubmitted` callback

File: example/simpleDemo/flutter_project/demo/lib/main.dart [hunk 1]
```diff
@@ -82,14 +84,30 @@ class _MyHomePageState extends State<MyHomePage> {
...
+                  onSubmitted: (value) {
+                    setState(() {
+                      _submittedMsg = value;
+                    });
+                  },
+                  onEditingComplete: () => FocusScope.of(context).requestFocus(myFocus),
                 ),
                 TextField(
                   decoration: InputDecoration(hintText: 'TextField 2'),
+                  maxLines: 2,
```
Comment (Drakirus) [inline example/simpleDemo/flutter_project/demo/lib/main.dart]: Added a multiLine TextField for testing purposes

**Review by pchampio — APPROVED** [review]

**Comment by Drakirus** [comment]
:+1:

**Comment by james-lawrence** [comment]
awesome, thanks @TitouanT

**Pull request merged**